core/src/main/scala/chisel3/Data.scala

```diff
@@ -665,6 +665,29 @@ abstract class Data extends HasId with NamedComponent with SourceInfoDoc {
     }
   }
 
+  private[chisel3] def legacyChiselConnect(
+    that: Data
+  )(
+    implicit sourceInfo:   SourceInfo,
+    connectCompileOptions: CompileOptions
+  ): Unit = {
+    if (connectCompileOptions.checkSynthesizable) {
+      requireIsHardware(this, s"data to be bulk-connected")
+      requireIsHardware(that, s"data to be bulk-connected")
+      (this.topBinding, that.topBinding) match {
+        case (_: ReadOnlyBinding, _: ReadOnlyBinding) => throwException(s"Both $this and $that are read-only")
+        // DontCare cannot be a sink (LHS)
+        case (_: DontCareBinding, _) => throw BiConnect.DontCareCantBeSink
+        case _ => // fine
+      }
+    }
+    if (!BiConnect.canBulkConnectData(this, that, sourceInfo, connectCompileOptions, Builder.referenceUserModule)) {
+      Builder.error(s"Cannot strict connect $this :<>= $that")
+    }
```

Comment:
I don't understand the function naming here. If this requires bulk connectable Data, this is definitely not a `legacyChiselConnect`.

Reply:
we have way too many terms here. Firrtl has "connect (<=)" and "partial connect (<-)". What is "bulk connect" and do we need that term at all in Chisel.

Reply:
The `BulkConnect` class in Chisel's internal IR is a misnomer because it actually corresponds to FIRRTL partial connect. In the case of `BiConnect.canBulkConnectData`, it is referring to a true "bulk connect" in the sense that it is a single FIRRTL connect that connects two Aggregates. Notably Adam is changing the name of that function in this PR from `canBulkConnectAggregates` to `canBulkConnectData` so perhaps we should come up with a better name if we're wanting to define it between two `Data`. `canIssueFIRRTLConnect` maybe?

Reply:
i think the check of `canIssueFirrtlConnect` is good. And I think the function called by `:<>=` is something like "directionedBulkConnect"

Reply:
I renamed this to `directionalBulkConnect`